### DISCUSSION TITLE
Demo site static build?

### DISCUSSION THREAD

**Discussion by eballeste** [pmndrs/zustand#1496] [Q&A] [answered]
Apologies if this is noobish and I am overstepping with my sorta unrelated question but I was wondering how is the example site for zustand deployed? I thought it was just built files but I noticed that the server has the whole dev directory within a static folder??? Is the browser doing the webpack build process??? If yes, how did you figure out how to do this? Any pointers? I would love to do something similar where I want to use zustand/zundo in a chrome browser extension project without having to build the script files locally using something like webpack. 

my main motivation for this question is that i'm using both my pc and my mac laptop for dev which is why i was originally planning on maintaining the state of the app myself, old school style, through plain js objects... but was thinking that i would need some sort of undo state mechanism which is how i landed on zustand/zundo as a possible tool to use.

**Comment by eballeste** [comment]
I guess my best-case scenario would be if there is a way to just load zustand as a global variable in the browser without the use of a node.js bundler. Is this possible with zustand or any of your other state management tools?

**Reply by dai-shi** [reply, marked as answer]
It's possible with modern browsers. Here's an example with jsfiddle.

```jsx
import { createElement, useState } from "https://esm.sh/react@18.2.0";
import { createRoot } from "https://esm.sh/react-dom@18.2.0";
import create from "https://esm.sh/zustand@4.1.5";

const useCount = create((set) => ({
  count: 0,
  inc: () => set((state) => ({ count: state.count + 1 })),
}));

const Counter = () => {
  const { count, inc: onClick } = useCount();
  return createElement(
    'div', {}, 'count with Zustand: ', count,
    createElement('button', { onClick }, '+1')
  )
}

const Counter2 = () => {
  const [count, setCount] = useState(0);
  const onClick = () => setCount((c) => c + 1);
  return createElement(
    'div', {}, 'count with useState: ', count,
    createElement('button', { onClick }, '+1')
  )
}

const App = () => {
  return createElement('div', {},
    createElement(Counter),
    createElement(Counter),
    createElement(Counter2),
    createElement(Counter2)
  )
}

createRoot(document.getElementById('app')).render(createElement(App));
```
https://jsfiddle.net/t7pjsfeb/

**Reply by eballeste** [reply]
amazing, thanks!

**Reply by eballeste** [reply]
my mind is blown with https://esm.sh/, thanks for the link!
